pkg/archive/copy.go

```diff
@@ -151,11 +154,19 @@ func CopyInfoSourcePath(path string, followLink bool) (CopyInfo, error) {
 		return CopyInfo{}, err
 	}
 
+	var dsize int64
+	if stat.Mode().IsDir() {
+		if dsize, err = directory.Size(resolvedPath); err != nil {
+			return CopyInfo{}, err
+		}
+	}
 	return CopyInfo{
 		Path:       resolvedPath,
 		Exists:     true,
 		IsDir:      stat.IsDir(),
 		RebaseName: rebaseName,
+		Size:       stat.Size(),
+		DSize:      dsize,
```

Comment:
Both places that use this information ignore `Size` and use `DSize` instead if the path is a directory, so it seems like it would be simpler to only have one `Size` variable.